docs_app/content/tutorial/8-subjects.md

```diff
@@ -0,0 +1,304 @@
+
+```ts
+interface OP_Observer<T> {
+    /** Handle a value notification */
+    next(value: T): void;
+}
+
+interface OP_Subject<T> extends OP_Observer<T> {
+    /** 
+     * Register an observer to be notified by this subject, and return
+     * a Subscription that will remove the observer when you call `unsubscribe()` on it.
+     */
+    subscribe(observer: OP_Observer<T>): Subcription;
+}
+```
+
+By adding a few additional methods we have exactly the same thing as what we have in RxJS:
+
+```ts
+interface Observer<T> {
+    /** Handle a value notification */
+    next(value: T): void;
+    /** Handle an error notification */
+    error(err: any): void;
+    /** Handle completion */
+    complete(): void;
+}
+
+interface Observable<T> {
+    /** 
+     * Register an observer to be notified, and return
+     * a Subscription that will remove the observer when you call `unsubscribe()` on it.
+     */
+    subscribe(observer: Observer<T>): Subcription;
+}
+
+interface Subject<T> extends Observable<T>, Observer<T> {
+}
+```
+
+## Subjects: Multicasting - The Primary Use Case
+
+If the primary use case isn't apparent from the APIs described above, it can be best reiterated as follows: Subjects are a type where many observers can be notified of the same value simulatenously. The primary use case of a subject is [multicasting](GL).
+
+In terms of RxJS and observables, this might still not seem clear. But think about it, if a subject is an [`Observer`](API), then it can be used to subscribe to an [`Observable`](API) by being passed to [`subscribe`](API). And if, when you call `next` on the subject, it notifies all observers who have subscribed to it (thus "registering" themselves to be "notified", per the [observer pattern](#subjects-are-from-the-observer-pattern)), then a [`Subject`](API) can be used to [multicast](GL) any observable like so:
+
+
+### Example: Basic Multicast With A Subject
+
+```ts
+// Create a subject
+const subject = new Subject();
+
+// Subscribe our consumers to the subject, not the source
+subject.subscribe(observerA);
+subject.subscribe(observerB);
+subject.subscribe(observerC);
+
+// Connect the source, through a single subscription, to all consumers by
+// pushing its values through the subject.
+const connectionSubscription = source$.subscribe(subject);
+```
+
+Note that we subscribe to the `subject` _before_ we connect `source$` through the `subject`. This is to ensure that if `source$` has any synchronous emissions, they are pushed to our [consumers](GL).
+
+## Subjects: Eventing - The Secondary Use Case
+
+The secondary use case for [`Subject`](API) and its kin, is to act as a way create an observable that we imperatively push notifications through later. This is similar to the use case of "deferreds" and promises.
```

Comment:
Regarding "and its kin", there was no prior mention of there being multiple types of subjects in this chapter. Should there be a section here on there being multiple types of Subjects? If not, would the newer reader have any idea what "and its kin" references? Consider either mentioning early on that there are multiple classes of Subject OR consider removing "and its kin" here and make this section just about "Subject" in general.